services/ui_backend_service/api/data_refiner.py

```diff
@@ -86,7 +91,7 @@ def _process(item):
             item['status'] = 'failed' if item['status'] == 'completed' and item['task_ok'] is False else item['status']
             item.pop('task_ok', None)
 
-            if item['foreach_stack']:
+            if item['foreach_stack'] and len(item['foreach_stack']) > 0 and len(item['foreach_stack'][0]) >= 4:
```

Comment:
If the `_postprocess` is skipped due to a feature flag, the `foreach_stack` will remain as an S3 location string. Is the logic on the length warranted in this case, even if it works as intended?